<?xml version="1.0"?>
<launch>
  <arg name="rviz" default="true" />

<!-- PARAMETERS -->
  <param name="use_sim_time" value="true" />

<!-- Export / Import directories -->
  <param name="directory_output" value="/home/$(env USER)/Downloads/export/" /> <!-- for both liosam and floam -->
  <param name="gnss/gps_logfile_path" value="/home/$(env USER)/Documents/2023-12-18_11-41/2023-12-18_11-41_gcp.csv" /> <!-- for both liosam and floam -->

  <!--arg name="slam_param_file" default="$(find lio_sam)/config/params_structor_indoor.yaml" /-->
  <arg name="slam_param_file" default="$(find lio_sam)/config/params_structor_outdoor.yaml" />
  <param name="map_resolution" type="double" value="0.5" />
  <param name="calib_ex" type="double" value="-0.24" />
  <param name="calib_ey" type="double" value="0.18" />
  <param name="calib_ez" type="double" value="180.0" />

<!-- SLAM EXPORTS -->
  <param name="export_slam_pcd" type="bool" value="true" />
  <param name="max_dis" type="double" value="90.0" />
  <param name="min_dis" type="double" value="0.5" />

<!-- Launching -->
  <node pkg="floam" type="floam_odom_estimation_node" name="floam_odom_estimation_node" output="log"/>
  <node pkg="floam" type="floam_laser_processing_node" name="floam_laser_processing_node" output="log"/>

  <include file="$(find lio_sam)/launch/slam_gps.launch">
    <arg name="slam_param_file" value="$(arg slam_param_file)" />
  </include>

  <include file="$(find structor_mapping)/launch/gps_utils.launch" /> <!-- only for testing-->
  
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/slam.rviz" />
  </group>
    
  

</launch>
